addons/pos_paytm/static/src/js/payment_paytm.js

```diff
@@ -0,0 +1,159 @@
+        const paymentLine = this.pos.get_order()?.selected_paymentline;
+        const order = this.pos?.selectedOrder;
+        let retry = localStorage.getItem(order.uid) || 0;
+        if (retry === 0) {
+            localStorage.setItem(order.uid, retry);
+        }
+        let transactionId = order.name.replace(' ', '').replaceAll('-', '').toUpperCase();
+        if (retry > 0) {
+            transactionId = transactionId.concat('retry', retry);
+        }
+        const transactionAmount = paymentLine.amount*100;
+        const timeStamp = Math.floor(Date.now() / 1000);
+
+        // Preparing Unique Random Reference Id
+        const referencePrefix = this.pos.config?.name.replace(/\s/g, "").slice(0,4)
+        const referenceId =  referencePrefix.concat(Math.floor(Math.random()*1000000000))
+        const response = await this.makePaymentRequest(transactionAmount, transactionId, referenceId, timeStamp);
+        if (response?.resultCode === "F") {
+            this._showError(response?.resultMsg);
+            paymentLine.set_payment_status('force_done');
+            this._incrementRetry(order.uid);
+            return false
+        }
+        paymentLine.set_payment_status('waitingCard');
+        const pollResponse = await this.pollPayment(transactionId, referenceId, timeStamp);
+        if (pollResponse) {
+            localStorage.removeItem(order.uid);
+            return true;
+        }
+        else {
+            this._incrementRetry(order.uid);
+            return false;
+        }
+    }
+
+    /**
+     * @Override
+     * @param {} order
+     * @param { string } cid
+     * @returns Promise
+     */
+    async send_payment_cancel(order, cid) {
+        await super.send_payment_cancel(...arguments);
+        const paymentLine = this.pos.get_order()?.selected_paymentline;
+        paymentLine.set_payment_status('retry');
+        this._incrementRetry(order.uid);
+        clearTimeout(this.pollTimeout);
+        return true;
+    }
+
+    /**
+     * @param { string } transactionId
+     * @param { string } referenceId
+     * @param { datetime } timestamp
+     * @returns Promise
+    */
+    async pollPayment(transactionId, referenceId, timestamp) {
+        const fetchPaymentStatus = async (resolve, reject) => {
+            const paymentLine = this.pos.get_order()?.selected_paymentline;
+            if (!paymentLine || paymentLine.payment_status == 'retry') {
+                return false;
+            }
+            try {
+                const data = await this.env.services.orm.silent.call(
+                    'pos.payment.method',
+                    'paytm_fetch_payment_status',
+                    [[this.payment_method.id], transactionId, referenceId, timestamp],
+                );
+                if (data?.error) {
+                    throw data?.error;
+                }
+                let resultCode = data?.resultCode;
+                if (resultCode === "F") {
```

Comment:
this error handling of "F" looks a lot like the one here above
maybe you need check this resultCode in this function _paytm_make_request and forward error to the JS